vengine/vengine/graphics/UIRenderer.cpp

```diff
@@ -164,60 +169,111 @@ void UIRenderer::setBitmapFont(
-            this->currentElementIndex * 6 - this->uiDrawCallData[this->uiDrawCallData.size() - 1].startVertex;
-    }
+        // Add data for unique draw call
+        UIDrawCallData drawCallData{};
+        drawCallData.textureIndex = textureIndex;
+        drawCallData.startVertex = this->currentElementIndex * 6;
 
-    // Set draw call
-    this->uiDrawCallData.push_back(drawCallData);
+        // Set number of vertices for previous draw call
+        if (this->uiDrawCallData.size() > 0)
+        {
+            // Num vertices for last draw call
+            this->uiDrawCallData[this->uiDrawCallData.size() - 1].numVertices =
+                this->currentElementIndex * 6 - this->uiDrawCallData[this->uiDrawCallData.size() - 1].startVertex;
+        }
+
+        // Set draw call
+        this->uiDrawCallData.push_back(drawCallData);
+        this->uiTextureIndex = textureIndex;
+    }
 }
 
 void UIRenderer::renderTexture(
-    const float& x, 
-    const float& y, 
-    const float& width, 
-    const float& height,
-    const uint32_t& u0,
-    const uint32_t& v0,
-    const uint32_t& u1,
-    const uint32_t& v1)
+    const glm::vec2& position,
+    const glm::vec2& dimension,
+    const glm::uvec4 textureCoords)
 {
     if (this->currentElementIndex >= START_NUM_MAX_ELEMENTS)
     {
         Log::error("Reached maximum number of rendered ui elements.");
         return;
     }
 
-    // Set element data
-    this->uiElementData[this->currentElementIndex].transform =
-        ResTranslator::transformRect(x, y, width, height);
-    this->uiElementData[this->currentElementIndex].uvRect =
-        glm::uvec4(u0, v0, u1, v1);
+    glm::vec2 extent = dimension / 2.0f;
+    glm::vec2 resTranslatorExtent = glm::vec2(ResTranslator::INTERNAL_WIDTH >> 1, ResTranslator::INTERNAL_HEIGHT >> 1);
+    if (position.x - extent.x < resTranslatorExtent.x && position.x + extent.x > -resTranslatorExtent.x &&
+        position.y - extent.y < resTranslatorExtent.y && position.y + extent.y > -resTranslatorExtent.y) // Within UI window
+    {
+        // Set element data
+        this->uiElementData[this->currentElementIndex].transform =
+            ResTranslator::transformRect(position, dimension);
+        this->uiElementData[this->currentElementIndex].uvRect =
+            textureCoords;
+
+        // Next ui element
+        this->currentElementIndex++;
+    }
+}
 
-    // Next ui element
-    this->currentElementIndex++;
+void UIRenderer::renderTexture(
+    const glm::vec3& position,
+    const glm::vec2& dimension,
+    const glm::uvec4 textureCoords)
+{
+    Scene* scene = this->sceneHandler->getScene();
+    if (!scene->entityValid(scene->getMainCameraID())) { return; }
+
+    Camera* cam = scene->getMainCamera();
+    Transform& camTransform = scene->getComponent<Transform>(scene->getMainCameraID());
+    cam->updateMatrices(camTransform);
+
+    glm::vec4 pos = cam->viewAndProj * glm::vec4(position, 1.0f);
+    if (pos.z > pos.w || pos.z <= 0) { return; }
```

Comment:
Should be "... || pos.z < 0" to be completely correct. The w-value is not 0 when z is 0 in NDC.